### ISSUE TITLE
PyPi Module

### ISSUE THREAD

**Issue by ChasNelson1990** [flatironinstitute/CaImAn#509]
Hi,

### Background:

I recently wanted to install caiman into my pipenv environment (and associated pipfile). I don't (and don't want to) use conda.

I'm able to create a pipenv inside the repo folder and use the environment.yml file to create a Pipfile that works inside the repo folder (installs all dependencies) and all unit tests (but the one requiring dowloaded data) succeed.

But I'm struggling to be able to install CaImAn itself from the Pipfile. It seems to be an issue with the setup.py but I really can't work it out (not an expert here). I think it's something todo with conda specific features of setup.py? And, to be fair, it's my choice not to use Conda (so this is not my issue).

### Here is my issue/suggestion:

However, it got me wondering why you've not chosen to add this to pypi? I feel like that would work with anybody's existing python pipeline (i.e. with or without conda).

If it's something you plan to do, happy to help if I can.

Chas

**Comment by pgunn** [comment]
Hi,
At least right now Conda gives us the ability to control the environment to a much greater degree than pip would (in a more cross-platform way), and it's easier to set up in a uniform way. I can't say we'd never consider adding to Pypi in the future, but right now it'd take a lot of extra work and make it harder for us to support users and continue to improve the software. It's not a matter of getting it done once so much as supporting it going forward on all three operating systems we currently support (our team is quite small). Providing a prebuilt conda package is a little more likely in the mid-term than Pypi support.

Our setup.py has nothing specific to conda. In theory, if you've set up a virtualenv that exactly mirrors what conda would give you, providing the right flags to pip would make it work (although I have not tried this).

**Comment by ChasNelson1990** [comment]
Thanks for replying. I'm not sure I appreciate the subtleties of conda vs pip to really see why supporting a pip module (once up and running) would be much different from supporting a conda package; however, it sounds like you've considered this so I'll trust your reasoning.

Okay. So, I started from scratch today to see if I can recreate (and/or fix) my problem and report back.

I can definitely recreate it but I've created it as a separate issue #510  with a pipenv related title (in case you want to keep this one as a future enhancement).
